### PR TITLE
Controlled selection

### PR DESCRIPTION
**Pull request by educastellano**
I implemented an optional controlled selection, based on the discussion in #60. 

Instead of `if/else` all the code, I made a `getState()` and `_setState()` which returns/sets the selection from/to the state or props depending on whether the `selected` prop is defined. If `selected` is not passed, the component behaves as usual. 

Also, I skipped returning the cell in the `onSelect()` callback, since it's not necessary and could be confusing if there is more than one cell selected. Instead I just pass `{ start, end }`. A single cell can be easily accessed by: 
```
onSelect({ start, end}) {
  const cell = data[start.i][start.j]
  ...
}
```

Let me know what you guys think about this implementation.

### CODE REVIEW

**Comment by nadbm** [comment]
Looks great @educastellano, I will merge this very soon this week alongside the typescript definitions.

File: lib/DataSheet.js [hunk 1]
```diff
@@ -603,6 +669,8 @@ DataSheet.propTypes = {
   onChange: _propTypes2.default.func,
   onCellsChanged: _propTypes2.default.func,
   onContextMenu: _propTypes2.default.func,
+  onSelect: _propTypes2.default.func,
+  selected: _propTypes2.default.object,
```
Comment (nadbm) [inline lib/DataSheet.js]: You can define a shape here for better type checking : 
https://reactjs.org/docs/typechecking-with-proptypes.html
```
optionalObjectWithShape: PropTypes.shape({
    color: PropTypes.string,
    fontSize: PropTypes.number
  }),
```

File: README.md [hunk 2]
```diff
@@ -199,6 +199,8 @@ rowRenderer | func | Optional function or React Component to render each row ele
 cellRenderer | func | Optional function or React Component to render each cell element. The default renders a `td` element.
 valueViewer | func | Optional function or React Component to customize the way the value for each cell in the sheet is displayed. Affects every cell in the sheet. See [cell options](https://github.com/nadbm/react-datasheet#cell-options) to override individual cells.
 dataEditor | func | Optional function or React Component to render a custom editor. Affects every cell in the sheet. Affects every cell in the sheet. See [cell options](https://github.com/nadbm/react-datasheet#cell-options) to override individual cells.
+selected | object | Optional. Wheter the selection is controlled or uncontrolled. Must be an object of this format: `{ start: { i: number, j; number }, end: { i: number, j: number } }`, or `null` for no selection.
```
Comment (nadbm) [inline README.md]: Whether *

**Comment by educastellano** [comment]
@nadbm Great! I fixed your reviews too.

**Pull request merged**